ts/components/leftpane/ActionsPanel.tsx

```diff
@@ -238,6 +239,16 @@ export const ActionsPanel = () => {
     return () => clearTimeout(timeout);
   }, []);
 
+  const globalUnreadMessageCount = useSelector(getGlobalUnreadMessageCount);
+  const unreadToShow = globalUnreadMessageCount;
+
+  // Reuse the unreadToShow from the global state to update the badge count
+  useEffect(() => {
+    if (unreadToShow !== undefined) {
+      ipcRenderer.send('update-badge-count', unreadToShow);
+    }
+  }, [unreadToShow]);
```

Comment:
you should use useThrottleFn for this, so we don't send 100 events per second when we fetch 100 messages in per seconds

Reply:
https://github.com/streamich/react-use/blob/master/docs/useThrottleFn.md